### PR TITLE
Pilot's Useless Fiesta; Part 1: The hunting trip

### PR DESCRIPTION
**Pull request by CrAzYPiLoT-SS13**
How many times has a space bear named Boris seen you, while you were trying to catch it? Well, this product fixes that problem! Adds the ability to attach signalers to bear traps to make them notify you when they catch something while you are hiding in your office. Happy hunting!

:cl: CrAzYPiLoT
rscadd: Adds the ability to attach signalers to bear traps. Stealthy hunting!
/:cl:

### CODE REVIEW

File: code/game/objects/items/weapons/legcuffs.dm [hunk 1]
```diff
@@ -60,12 +67,33 @@
...
+			return
+		if(sig)
+			to_chat(user, "<span class='warning'>This beartrap already has a signaler hooked up to it!</span>")
+			return
+		sig = I
+		if(sig.secured)
+			to_chat(user, "<span class='notice'>The signaler is secured.</span>")
+			sig = null
+			return
+		user.drop_item(src)
```
Comment (Tastyfish) [inline code/game/objects/items/weapons/legcuffs.dm]: `drop_item` doesn't take an argument.

Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/weapons/legcuffs.dm]: The IED does it with an argument, it works, and, I cannot change that ATM. I don't think it's that bug of a problem, as it's consistent.

Comment (TheDZD) [inline code/game/objects/items/weapons/legcuffs.dm]: [Read the aggressive comment](https://github.com/ParadiseSS13/Paradise/blob/master/code/modules/mob/inventory.dm#L91).

Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/weapons/legcuffs.dm]: Alrighty then.

File: code/game/objects/items/weapons/legcuffs.dm [hunk 2]
```diff
@@ -60,12 +67,33 @@
 				to_chat(user, "<span class='danger'>You shouldn't be reading this message! Contact a coder or someone, something broke!</span>")
 				IED = null
 				return
+	if(istype(I, /obj/item/device/assembly/signaler))
+		if(IED)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/legcuffs.dm]: why does it matter about the IED?

File: code/game/objects/items/weapons/legcuffs.dm [hunk 3]
```diff
@@ -60,12 +67,33 @@
...
+			to_chat(user, "<span class='warning'>This beartrap already has an IED hooked up to it!</span>")
+			return
+		if(sig)
+			to_chat(user, "<span class='warning'>This beartrap already has a signaler hooked up to it!</span>")
+			return
+		sig = I
+		if(sig.secured)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/legcuffs.dm]: why is this here?

File: code/game/objects/items/weapons/legcuffs.dm (hunk 2 above)
Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/weapons/legcuffs.dm]: I wanted it to not be addable if there is something already attached to it, due to the sizes of the IED and the bear trap.

File: code/game/objects/items/weapons/legcuffs.dm (hunk 3 above)
Comment (CrAzYPiLoT-SS13) [inline code/game/objects/items/weapons/legcuffs.dm]: As with most assemblies, you first need to screwdriver the signaler before attaching it.
